ttnn/cpp/ttnn/tensor/tensor_ops.cpp

```diff
@@ -391,7 +391,12 @@ Tensor tensor_reshape(const Tensor& input_tensor, const ttnn::Shape& new_shape)
                     if (tensor.memory_config().memory_layout != TensorMemoryLayout::HEIGHT_SHARDED) {
                         DeviceStorage device_storage = std::get<T>(tensor.get_storage());
                         DeviceBuffer device_buffer = device_storage.get_buffer();
-                        device_buffer->set_page_size(new_shape[-1] * tensor.element_size());
+                        if (new_shape[-1] != 0) {
+                            device_buffer->set_page_size(new_shape[-1] * tensor.element_size());
+                        }
+                        else {
+                            device_buffer->set_page_size(2 * tensor.element_size());
```

Comment:
nit: 2 is a magic number. Would be good to replace with a meaningful variable/constant.